Dockerfile.node

```diff
@@ -0,0 +1,58 @@
+FROM crystallang/crystal:0.35.1-alpine as crystal
+
+WORKDIR /app
+COPY node ./node
+COPY cli ./cli
+COPY types ./types
+COPY client ./client
+COPY volgen ./volgen
+
+RUN cd node && shards install --production
+RUN cd node && VERSION=${VERSION} shards build --static --release --stats --time
+
+RUN cd cli && shards install --production
+RUN cd cli && VERSION=${VERSION} shards build --static --release --stats --time
+
+FROM ubuntu:20.04 as prod
+
+ENV DEBIAN_FRONTEND=noninteractive
+ENV PATH="/kadalu/bin:/opt/bin:/opt/sbin:$PATH"
+COPY --from=kadalu-storage/builder:latest /opt /opt
+
+RUN apt-get update -yq && \
+    apt-get install -y --no-install-recommends python3 xfsprogs libtirpc3 init && \
+    apt-get -y clean && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN mkdir -p /kadalu/templates /kadalu/volfiles
+RUN mkdir -p /var/run/gluster /var/log/glusterfs
+
+RUN mkdir -p /var/lib/kadalu /var/run/kadalu /var/lib/kadalu/volfiles /var/log/kadalu
+COPY extra/kadalu-brick@.service /lib/systemd/system/
+COPY extra/kadalu-node.service /lib/systemd/system/
+COPY extra/kadalu-brick /usr/sbin/
+RUN chmod +x /usr/sbin/kadalu-brick
+
+COPY --from=crystal /app/node/bin/kadalu-node /usr/sbin/kadalu-node
+COPY --from=crystal /app/cli/bin/kadalu /usr/sbin/kadalu
+
+RUN systemctl enable kadalu-node
```

Comment:
asking bit late I guess. Which is quicker? kadalu-node as binary handling glusterfs docker images? or a single docker image with node and everything combined?

Reply:
Found this approach simple. With this approach we can spawn multiple containers in the same machine as required. each container will become node/brick. It is also possible to export multiple bricks from same container.